packages/ai/src/Button.ts

```diff
@@ -58,13 +87,185 @@ class Button extends UI5Element {
 	@property({ type: Boolean })
 	disabled!: boolean;
 
+	/**
+	 * Defines the current state of the component.
+	 *
+	 * **Note:** if nothing is defined, the component will be set initially to the first defined state (if any).
+	 *
+	 * @default ""
+	 * @public
+	 */
 	@property()
-	icon!: string;
+	state?: string;
 
-	@property()
-	endIcon!: string;
+	/**
+	 * Keeps the current state of the component.
+	 * @private
+	 */
+	@property({ type: Object })
+	_currentStateObject?: ButtonState;
+
+	/**
+	 * Initiates button elements fade-out phase.
+	 * @default false
+	 * @private
+	 */
+	@property({ type: Boolean })
+	fadeOut!: boolean;
+
+	/**
+	 * Initiates button fade middle phase.
+	 * @default false
+	 * @private
+	 */
+	@property({ type: Boolean })
+	fadeMid!: boolean;
+
+	/**
+	 * Initiates button elements fade-in phase.
+	 * @default false
+	 * @private
+	 */
+	@property({ type: Boolean })
+	fadeIn!: boolean;
+
+	/**
+	 * Defines the available states of the component.
+	 * **Note:** Although this slot accepts HTML Elements, it is strongly recommended that you only use `ui5-ai-button-state` components in order to preserve the intended design.
```

Comment:
Maybe that's related to the type, so we can change the type to HTMLElement

Reply:
Yes, but in that case someone can put non-ButtonState elements inside, which will not work.